contracts/coordinator/src/index.ts

```diff
@@ -1,5 +1,41 @@
-export * from './artifacts';
-export * from './wrappers';
+export { artifacts } from './artifacts';
+export {
+    CoordinatorContract,
+    CoordinatorRegistryContract,
+    LibConstantsContract,
+    LibCoordinatorApprovalContract,
+    LibCoordinatorRichErrorsContract,
+    LibEIP712CoordinatorDomainContract,
+} from './wrappers';
 export import CoordinatorRevertErrors = require('./revert_errors');
 export { ApprovalFactory } from './approval_factory';
 export { SignedCoordinatorApproval } from './types';
+export { SignatureType, SignedZeroExTransaction, EIP712DomainWithDefaultSchema } from '@0x/types';
+export {
```

Comment:
Why is this being exported in both `@0x/contracts-dev-utils` and this package? Do we only expect most consumers to use one of these?

Reply:
This actually seems to be the case for many contract packages. Do we just want the important bits of `ethereum-types` to be accessible from all of the contracts packages?

Reply:
@jalextowle it's actually more complicated then that.

First, unlike vanilla `tsdoc`, our docs only render things exported from `index.ts`. This lets us define what the public interface of our package is and hide the complexities of it's internal from the end developer.

In order to support this behavior, we render the docs for the entire package with `tsdoc` and  then remove anything not exported in `index.ts`. For all things exported in `index.ts`, we also remove private methods/properties, etc... For all that is left, we then make sure all parameter & return value types are also exported. Why? So that the docs are comprehensive, and we don't reference a type for which the docs don't render a corresponding declaration. Imagine a method referencing type `LotsOfOpts` without defining what those opts are. Second, it makes sure the end developer can define `LotsOfOpts` and then pass it into the function. Third, imagine a class that takes a `Web3ProviderEngine` instance as a constructor arg but the package doesn't export the `Web3ProviderEngine` class. Our checks will catch this as well and make sure a package has all the requisite exports to make the interface usable.

Without these checks, we'd be publishing a bunch of libraries that expect specific types in their interfaces without making sure the packages _also_ provide those types to the end developer.

Reply:
Thanks for the thoughtful reply. That makes sense! 

@fabioberger Also explained offline that duplicated exports from multiple packages are de-duplicated if they are of the same version, which answers a follow-up to these questions.